### PR TITLE
CF-3001 - Please make "Skip tutorial" a config option

### PR DESCRIPTION
**Pull request by aldrichdev**
- Added `show_tutorial_skip_option` config option for whether to show "Skip the tutorial" menu button (true by default).
- Tested all possible variations of `member_world` & `show_tutorial_skip_option` locally, no issues encountered clicking logout button, tutorial button either appears and is clickable or does not and clicking where it was does nothing.

### CODE REVIEW

**Comment by aldrichdev** [comment]
Needs a few more finishing touches, just noticed this. The menu height is different off of tutorial island.
[](https://cdn.discordapp.com/attachments/421836426217521154/800823125683077158/unknown.png)

**Comment by aldrichdev** [comment]
Fixed issues with vertical alignment by sending 0 for `inTutorial()` if `show_tutorial_skip_option` is false, instead of checking the config value in `mudclient.java`. Prevent server-side from working if it false as well in case someone hacks the client or something. Confirmed this works whether the new config option is true or false; skip button works if true, boatman still works if false.

**Comment by Hubcapp** [comment]
It basically looks fine, but "showTutorialSkipOption" doesn't seem to be used anywhere and should be removed. If you're able to revert the hundreds of lines of changed code so that there are no changes related to that integer that would be great.

File: server/src/com/openrsc/server/net/rsc/ActionSender.java [hunk 1]
```diff
@@ -659,61 +660,62 @@ static void sendServerConfigs(Player player) {
 		s.writeByte((byte) (server.getConfig().SHOW_ROOF_TOGGLE ? 1 : 0)); // 21
 		s.writeByte((byte) (server.getConfig().WANT_HIDE_IP ? 1 : 0)); // 22
 		s.writeByte((byte) (server.getConfig().WANT_REMEMBER ? 1 : 0)); // 23
-		s.writeByte((byte) (server.getConfig().WANT_GLOBAL_CHAT ? 1 : 0)); // 24
```
Comment (Hubcapp) [inline server/src/com/openrsc/server/net/rsc/ActionSender.java:662]: revert this chunk

File: Client_Base/src/orsc/PacketHandler.java [hunk 2]
```diff
@@ -899,7 +899,7 @@ private void setServerConfiguration() {
 		int sideMenuToggle, inventoryCountToggle, zoomViewToggle;
 		int menuCombatStyleToggle, fightmodeSelectorToggle, experienceCounterToggle;
 		int experienceDropsToggle, itemsOnDeathMenu, showRoofToggle, wantHideIp, wantRemember;
-		int wantGlobalChat, wantSkillMenus, wantQuestMenus, wantQuestStartedIndicator, maxWalkingSpeed;
```
Comment (Hubcapp) [inline Client_Base/src/orsc/PacketHandler.java:902]: Should be able to just copy and paste in the old file, don't see any changes in here that need to be kept

**Comment by Hubcapp** [comment]
Would also ask that the order of those config bytes don't get changed (for future features that might require the inauthentic client config opcode), so there are less lines of code changing that need to be reviewed.

File: Client_Base/src/orsc/PacketHandler.java (hunk 2 above)
Comment (aldrichdev) [reply Client_Base/src/orsc/PacketHandler.java:902]: Hey Logg, I am working on this one again, and I don't quite understand what you mean by this.

**Comment by aldrichdev** [comment]
How do we reopen this PR? I don't see an option to do so. I did fix the byte order by putting the new config option at the very bottom of each list.

**Comment by Hubcapp** [comment]
There have been no new commits on CF-3001, so currently the feedback given in January still applies. Reopening anyway.

**Comment by Hubcapp** [comment]
Oh, it's a github quirk where it doesn't show new commits on closed PRs...

File: server/dev.conf [hunk 3]
```diff
@@ -0,0 +1,181 @@
+# Dev Testing World Config
```
Comment (Hubcapp) [inline server/dev.conf:1]: remove this file

File: server/src/com/openrsc/server/ServerConfiguration.java [hunk 4]
```diff
@@ -428,6 +429,7 @@ void initConfig(String defaultFile) throws IOException {
 		HIDE_LOGIN_BOX_TOGGLE = tryReadBool("hide_login_box_toggle").orElse(false);
 		BATCH_PROGRESSION = tryReadBool("batch_progression").orElse(false);
 		CUSTOM_IMPROVEMENTS = tryReadBool("custom_improvements").orElse(false);
+		SHOW_TUTORIAL_SKIP_OPTION = tryReadBool("show_tutorial_skip_option").orElse(false);
```
Comment (Hubcapp) [inline server/src/com/openrsc/server/ServerConfiguration.java:432]: `.orElse(true);` needed for old configs

**Review by Hubcapp — CHANGES_REQUESTED** [review]
Not sure what changed in `server/inc/sqlite/preservation.db`, but if you're able to, please revert changes on that file.

File: Client_Base/src/orsc/PacketHandler.java (hunk 2 above)
Comment (Hubcapp) [reply Client_Base/src/orsc/PacketHandler.java:921]: The client doesn't seem to actually use showTutorialSkipOption at any time, so it doesn't need to know about it, and the feature should still work. You're handling the disabling of the Tutorial Skip button entirely in packethandler.

File: server/src/com/openrsc/server/net/rsc/ActionSender.java [hunk 5]
```diff
@@ -807,6 +808,7 @@ private static Packet prepareServerConfigs(Server server) {
 		configs.add((byte) (server.getConfig().WANT_CERT_AS_NOTES ? 1 : 0)); // 79
 		configs.add((byte) (server.getConfig().WANT_OPENPK_POINTS ? 1 : 0)); // 80
 		configs.add((byte) (server.getConfig().OPENPK_POINTS_TO_GP_RATIO)); // 81
+		configs.add((byte) (server.getConfig().SHOW_TUTORIAL_SKIP_OPTION ? 1 : 0)); // 82
```
Comment (Hubcapp) [inline server/src/com/openrsc/server/net/rsc/ActionSender.java:811]: not needed once Client_Base file is reverted

File: Client_Base/src/orsc/PacketHandler.java (hunk 2 above)
Comment (Hubcapp) [reply Client_Base/src/orsc/PacketHandler.java:921]: i.e. you should be able to revert all Client_Base files back to the way they were before & still have identical functionality.

File: server/src/com/openrsc/server/net/rsc/ActionSender.java (hunk 5 above)
Comment (aldrichdev) [reply server/src/com/openrsc/server/net/rsc/ActionSender.java]: @Hubcapp This has been removed.

File: server/src/com/openrsc/server/ServerConfiguration.java (hunk 4 above)
Comment (aldrichdev) [reply server/src/com/openrsc/server/ServerConfiguration.java]: @Hubcapp This has been updated.

File: server/dev.conf (hunk 3 above)
Comment (aldrichdev) [reply server/dev.conf]: @Hubcapp Removed!

File: Client_Base/src/orsc/PacketHandler.java (hunk 2 above)
Comment (aldrichdev) [reply Client_Base/src/orsc/PacketHandler.java]: > Not sure what changed in `server/inc/sqlite/preservation.db`, but if you're able to, please revert changes on that file.

@Hubcapp Done - I have reverted that file to its state in `develop`.

Comment (aldrichdev) [reply Client_Base/src/orsc/PacketHandler.java]: "i.e. you should be able to revert all Client_Base files back to the way they were before & still have identical functionality."
@Hubcapp you're right, I thought those changes to `Client_Base` were needed for the menu to hide the button, but apparently not? Those changes have been reverted.

**Comment by aldrichdev** [comment]
@Hubcapp I tried reverting the commit that added `preservation.db` as a change and it did not work. I don't think the file actually change because the diff says 0 bytes were added, but let me know if I should create a new branch and manually apply all changes except to that file, and start a new PR.

**Comment by aldrichdev** [comment]
Closing as it was decided a new branch and PR should be created: https://github.com/Open-RSC/Core-Framework/pull/1832

**Pull request closed without merging**